### PR TITLE
[RNMobile] refine Block borders

### PR DESCRIPTION
**Pull request by jbinda**
## Description

PR goal is to apply `InnerBlock` bordering style to other blocks on the RootList level ( full solid border ) as well as show `FloatingToolbar` when selected.

At this moment two commits is the minimum code change to achieve expected results. I will also do the code cleanup to remove unnecessary code in future commits.

Please also refer to:
[Related gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1859)
[Discussion on design](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1802)

## How has this been tested?

1. Start with below code in `initial-html`

<details>

```
/**
 * @format
 * @flow
 */

export default `
<!-- wp:group -->
<div class="wp-block-group"><div class="wp-block-group__inner-container"><!-- wp:heading -->
<h2>Group</h2>
<!-- /wp:heading -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image --></div></div>
<!-- /wp:group -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->

<!-- wp:image -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->


`;
```

</details>

2. Try to play with selection on RootList block and the Group block
3. Add as many Blocks as you want to RootList to test how the modified border looks like

## Screenshots 

<image src="https://user-images.githubusercontent.com/21242757/73736749-9f57d000-4741-11ea-8d40-dfeafa56fddb.gif" width="400" />

## Types of changes

New feature ( apply same bordering style as in InnerBlock to RootList blocks )

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-list/index.native.js
Comment (jbinda) [inline packages/block-editor/src/components/block-list/index.native.js]: @etoledom I have refactor the bordering logic according to convo with @iamthomasbishop and make some clean-up in the code.

However I'm wondering how to approach on this line ( I see that commit description i to prevent crash after merging gutenberg master to mobile ).

`isFullyBordered` flag was true in two scenarios:
1. For Blocks that holds InnerBlocks inside (it was hardcoded to true in `InnerBlock` component)
2. When the screen size was Mobile (the login in Layout component hold this value in state and pass it down)

So the false goes only to the RootList in only one case - in vertical position of the screen.

I believe that the purpose of that prop is to make sure `FlatList` will be rerender after state change.

According to all above the question is do you think we can hardcode it to true like above or we need to add some logic ?

Comment (etoledom) [reply packages/block-editor/src/components/block-list/index.native.js]: Hey @jbinda , thank for asking!

A bit of context about that merge commit: It was a big PR that solved a breaking change on the `gutenberg` repo. It needed a fair bit of refactoring.

This snippet you mention is not related to the crash itself, but it just came along as part of the refactoring, so changing it shouldn't have risks of a crash regression.

> So the false goes only to the RootList in only one case - in vertical position of the screen.

This is the case not only for phones on portrait mode, but on any case where the editor width was small enough. (Other example is multitasking on iPads).

As I see that we now, we want to always drawing the side lines, no matter the editor width. If that's the case then it sounds good to keep it always as true, or maybe we could remove this line completely since we won't have visual changes to re-render.

Comment (koke) [reply packages/block-editor/src/components/block-list/index.native.js]: > maybe we could remove this line completely since we won't have visual changes to re-render

That's what I was thinking. From what I understand, `extraData` is only used to decide whether the component needs to re-render. So it doesn't care about the actual value, only if it changes. If we always pass `true`, it seems equivalent to not passing it at all.

Comment (jbinda) [reply packages/block-editor/src/components/block-list/index.native.js]: @etoledom @koke 

Thanks for reply. I thought so this line won't be necessary but want to make sure according to @etoledom commit message about the reason behind mentioned crash.

So the conclusion is I will remove this line.

Apart of that do you agree with the rest of  the code cleanup that I did ?

Comment (pinarol) [reply packages/block-editor/src/components/block-list/index.native.js]: > extraData={ [ isFullyBordered ] }

That line was only necessary to get the view updated when border style changes, in cases like switching between Landscape & Portrait. If it is going to be a constant value we don't need it anymore.

Comment (jbinda) [reply packages/block-editor/src/components/block-list/index.native.js]: according to above conversation line was removed

**Review by koke — APPROVED** [review]
Looking good. Yay for less code 😁

**Comment by pinarol** [comment]
so happy to see some complex logic removed! 🥳

**Pull request merged**
